clarity-ext-scripts/clarity_ext_scripts/covid/knm_service.py

```diff
@@ -0,0 +1,78 @@
+from clarity_ext_scripts.covid.partner_api_client import PartnerAPIV7Client
+from clarity_ext.utils import lazyprop
+
+
+def KNMConfig(config):
+    """
+    Creates config required for KNM from the clarity-ext config (which has more than that)
+    """
+    return {
+        key: config[key]
+        for key in [
+            "test_partner_base_url",
+            "test_partner_code_system_base_url",
+            "test_partner_user",
+            "test_partner_password"
+        ]
+    }
+
+
+def KNMClientFromExtension(extension):
+    # A factory for a KnmClient from an extension
+    config = KNMConfig(extension.config)
+    return PartnerAPIV7Client(**config)
+
+
+class KNMSampleAccessor(object):
+    """
+    Describes a sample/analyte in the LIMS that has originally come via the KNM workflow
+    """
+
+    def __init__(self, org_uri, org_referral_code, date_arrival):
+        self.org_uri = org_uri
+        self.org_referral_code = org_referral_code
+        self.date_arrival = date_arrival
+
+    @classmethod
+    def create_from_lims_sample(cls, sample):
+        pass
```

Comment:
Maybe remove it or make it raise `NotImplementedException`?

Reply:
Yes, this will be followed up right after this PR (we'll need it to be able to connect it with the LIMS)